src/components/ui/bar/StockSearch.jsx

```diff
@@ -1,52 +1,235 @@
+  flex-direction: column;
+  width: 100%;
+  padding-right: 0px;
+`;
+
+const StockName = styled.p`
+  font-size: 28px;
+  font-weight: bold;
+  margin: 0px;
+  color: #000;
+`;
+
+const StockText = styled.div`
+  display: flex;
+  width: 100%;
+`;
+
+const Kospi = styled.p`
+  font-size: 18px;
+  margin: 0;
+  margin-right: 5px;
+  font-size: 15px;
+  color: black;
+  font-weight: bold;
+`;
+
+const DateText = styled.p`
+  font-size: 18px;
+  margin: 0;
+  font-size: 15px;
+  color: black;
+  padding-top: 2px;
+  font-weight: bold;
+`;
+
+const PriceContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  margin-top: 0px;
+`;
+
+const PriceNumContainer = styled.div`
+  display: flex;
+  width: 100%;
+`;
+
+const Price = styled.p`
+  font-size: 28px;
+  font-weight: bold;
+  color: #ed3738;
+  margin: 0;
+`;
+
+const PriceChange = styled.p`
+  font-size: 15px;
+  color: ${({ positive }) => (positive ? "#ED3738" : "#f44336")};
+  margin: 0;
+  margin-right: 5px;
+  font-weight: bold;
+`;
+
+const PriceRatio = styled.p`
+  font-size: 15px;
+  color: ${({ positive }) => (positive ? "#ED3738" : "#f44336")};
+  margin: 0;
+  font-weight: bold;
+`;
+
 function StockSearch() {
-  const priceChange = 14000; // 예시 데이터: 가격 변화 값
-  const priceRatio = 4.54; // 예시 데이터: 현재 가격
+  const [searchTerm, setSearchTerm] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [stockList, setStockList] = useState([]);
+  const [selectedStock, setSelectedStock] = useState(null);
+  const priceChange = selectedStock ? selectedStock.priceChange : 14500;
+  const priceRatio = selectedStock ? selectedStock.priceRatio : 4.54;
+
+  const fetchStockList = async (query) => {
+    try {
+      // 실제 API 요청 코드 작성할 것임.
+      // const response = await axios.get(`/api/search-stocks?query=${query}`);
+      // setStockList(response.data.stocks); // assuming the response has a "stocks" array
+
+      // 로컬 환경을 위해 가짜 데이터 사용
+      setStockList(["삼성전자", "LG전자", "SK하이닉스", "현대자동차"]);
+    } catch (error) {
+      console.error("Error fetching stock list:", error);
+      // 로컬 환경을 위해 가짜 데이터 사용
+      setStockList(["삼성전자", "LG전자", "SK하이닉스", "현대자동차"]);
+    }
+  };
+
+  const fetchStockDetails = async (stockName) => {
+    try {
+      // 여기에 실제 API 요청 코드를 추가하세요.
+      // const response = await axios.get(`/api/stock-details?name=${stockName}`);
+      // setSelectedStock(response.data); // assuming the response has stock details
+
+      // 로컬 환경을 위해 가짜 데이터 사용
+      setSelectedStock({
+        name: stockName,
+        priceChange: 2000,
+        priceRatio: 2.5,
+        price: 120000,
+      });
+    } catch (error) {
+      console.error("Error fetching stock details:", error);
+      // 로컬 환경을 위해 가짜 데이터 사용
+      setSelectedStock({
+        name: stockName,
+        priceChange: 2000,
+        priceRatio: 2.5,
+        price: 120000,
+      });
+    }
+  };
+
+  const isKorean = (value) => {
+    const regex = /^[가-힣]+$/;

```

Comment:
국내주식중에서도 한글이 아닌 것들도 있을 거라, 알파벳도 허용해줘야 할 것 같아요.
그럼 이 함수는 이제 더이상 '한글이냐'를 의미하는 게 아니라 '완전한문자들이냐'를 의미하는 것일 거라 함수명도 좀 달라져야 할 것 같습니다.

```suggestion
  const isCompleteCharacters = (value) => {
    const regex = /^[가-힣a-zA-Z\s]+$/;
```